superset/connectors/druid/models.py

```diff
@@ -1612,7 +1612,7 @@ def query_datasources_by_name(
     ) -> List["DruidDatasource"]:
         return (
             session.query(cls)
-            .filter_by(cluster_name=database.id)
+            .filter_by(cluster_name=database.database_name)
```

Comment:
@michellethomas 
This didn't cause any problems for us because we use our own security manager, so this function isn't used. 

I did a little more digging, and I'm not entirely certain that `cluster_name=database.database_name` is correct, since clusters != databases in our tables. Maybe this line should just be removed? Do you know anyone with more context on this kind of thing? Either way, this change won't cause any (more) problems, but it does seem wrong.

Reply:
Maybe @mistercrunch could chime in here? This code is pretty old

Reply:
Mmmh, for reference, this came from #2497.

Clearly there are issues around the Connector interface. Good news is we're planning on deprecating the Druid connector in favor of SQLAlchemy

Reply:
This new line is likely to fail as much as the previous line.